pkg/ingester/errors.go

```diff
@@ -48,5 +48,5 @@ func (e *validationError) Error() string {
 
 // wrapWithUser prepends the user to the error. It does not retain a reference to err.
 func wrapWithUser(err error, userID string) error {
-	return fmt.Errorf("user=%s: %s", userID, err)
+	return fmt.Errorf("user=%s: %w", userID, err)
```

Comment:
By using `%w` here, we are violating documented contract of this function that says "It does not retain a reference to err.".